src/System.Private.Interpreter/src/Internal/Runtime/Interpreter/ILImporter.Interpreter.cs

```diff
@@ -239,22 +255,1021 @@ private void ImportLoadFloat(double value)
+                            {
+                                result = (ulong)val2 < (ulong)val1;
+                            }
+                        }
+                        else
+                        {
+                            ThrowHelper.ThrowInvalidProgramException();
+                        }
+                    }
+                    break;
+                case StackValueKind.NativeInt:
+                    {
+                        IntPtr val1 = op1.AsNativeInt();
+                        if (op2.Kind == StackValueKind.Int32
+                            || op1.Kind == StackValueKind.Int64
+                            || op2.Kind == StackValueKind.NativeInt)
+                        {
+                            IntPtr val2 = IntPtr.Zero;
+
+                            if (op2.Kind == StackValueKind.Int32)
+                            {
+                                val2 = new IntPtr(op2.AsInt32());
+                            }
+                            else if (op2.Kind == StackValueKind.Int64)
+                            {
+                                val2 = new IntPtr(op2.AsInt64());
+                            }
+                            else
+                            {
+                                val2 = op2.AsNativeInt();
+                            }
+
+                            if (opcode == ILOpcode.ceq)
+                            {
+                                result = val1 == val2;
+                            }
+                            else if (opcode == ILOpcode.cgt)
+                            {
+                                result = (ulong)val2.ToInt64() > (ulong)val1.ToInt64();
+                            }
+                            else if (opcode == ILOpcode.cgt_un)
+                            {
+                                result = ((UIntPtr)val2.ToPointer()).ToUInt64() > ((UIntPtr)val1.ToPointer()).ToUInt64();
+                            }
+                            else if (opcode == ILOpcode.clt)
+                            {
+                                result = (ulong)val2.ToInt64() < (ulong)val1.ToInt64();
+                            }
+                            else if (opcode == ILOpcode.clt_un)
+                            {
+                                result = ((UIntPtr)val2.ToPointer()).ToUInt64() < ((UIntPtr)val1.ToPointer()).ToUInt64();
+                            }
+                        }
+                    }
+                    break;
+                case StackValueKind.Float:
+                    {
+                        double val1 = op1.AsDouble();
+
+                        if (op2.Kind == StackValueKind.Float)
+                        {
+                            double val2 = op2.AsDouble();
+
+                            if (opcode == ILOpcode.ceq)
+                            {
+                                result = (double.IsNaN(val1) || double.IsNaN(val2)) ? false : val1 == val2;
```

Comment:
Do you really need to check for NaNs explicitly? I would expect the default operators to have the right behavior.